### ISSUE TITLE
multi-scale training?

### ISSUE THREAD

**Issue by suyanzhou626** [PyTorchLightning/pytorch-lightning#11506]
## 🚀 Feature

The `multi-scale training`  is a key trick in object detection and semantic segmentation, such as, [PraNet](https://github.com/DengPingFan/PraNet/blob/dfddf135d67db70b3fdee03c82e99916402ee8e4/MyTrain.py#L27), [Yolo](https://github.com/ultralytics/yolov5/blob/db1f83be6312a3d68c817fd25251194e5e7e5b5d/train.py#L320).

### Motivation

It can effectively improve the generalization ability of the model at multiple scales.

### Pitch

```
scales = [0.5,0.75,1.0,1.25,1.5,2.0]
for scale in scales:
    def training_step(self, batch, batch_idx):
        image, label = batch 
        if scale != 1:
             size = image.shape
             scale_size = size*scale
             image = upsample(image, scale size)
             label = upsample(label, scale_size)
        pred = self.model(image)
        loss = loss_func(pred, label)
        return {'loss': loss}
```
In training step, it only process a single image scale in a loss backpropagation.
We want to backpropagate the gradient in every scale instead of summing the loss together then backpropagation in `training_step`.

### Alternatives

Pure pytorch example can be seen in [PraNet](https://github.com/DengPingFan/PraNet/blob/dfddf135d67db70b3fdee03c82e99916402ee8e4/MyTrain.py#L27), [Yolo](https://github.com/ultralytics/yolov5/blob/db1f83be6312a3d68c817fd25251194e5e7e5b5d/train.py#L320).

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda

**Comment by akihironitta** [comment]
Hi @suyanzhou626, I believe you can do it with manual optimization. See the documentation for details: https://pytorch-lightning.readthedocs.io/en/1.5.10/common/optimizers.html
